packages/react/src/number-field/utils/parse.ts

```diff
@@ -31,15 +31,19 @@ export function getNumberLocaleDetails(
   return result;
 }
 
-export function parseNumber(formattedNumber: string, options?: Intl.NumberFormatOptions) {
-  let locale: string | undefined;
+export function parseNumber(
+  formattedNumber: string,
+  locale?: Intl.LocalesArgument,
+  options?: Intl.NumberFormatOptions,
+) {
+  let computedLocale = locale;
   if (ARABIC_RE.test(formattedNumber)) {
-    locale = 'ar';
+    computedLocale = 'ar';
```

Comment:
Do we really want to override this now that it is specified by the user?

Reply:
Yeah you're right, it should check for `undefined`